lib/types.js

```diff
@@ -2071,6 +2072,10 @@ function RecordType(schema, opts) {
   }, this));
   this._branchConstructor = this._createBranchConstructor();
   this._isError = schema.type === 'error';
+
+  // `Error.captureStackTrace` is only available in V8 environments
+  this._showStack = !opts.noStackTraces && typeof Error.captureStackTrace === 'function';
```

Comment:
I'd remove this and just pass in the option to `this._createConstructor` (see other comment above). The comment should go in the conditional block where `Error.captureStackTrace` is called.